File: graphql/graphql-thorntail/src/main/java/org/example/graphql/api/PeopleApi.java
package org.example.graphql.api;

import org.eclipse.microprofile.graphql.DefaultValue;
import org.eclipse.microprofile.graphql.Description;
import org.eclipse.microprofile.graphql.GraphQLApi;
import org.eclipse.microprofile.graphql.GraphQLException;
import org.eclipse.microprofile.graphql.Mutation;
import org.eclipse.microprofile.graphql.Name;
import org.eclipse.microprofile.graphql.Query;
import org.eclipse.microprofile.graphql.Source;
import org.example.graphql.database.Database;
import org.example.graphql.model.Gender;
import org.example.graphql.model.Person;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;
import java.util.Collection;
import java.util.UUID;
import java.util.logging.Level;

@GraphQLApi
@ApplicationScoped
public class PeopleApi {

    @Inject
    Database database;

    // To try out, see queries/query-all-persons* files
    @Query(value = "all")
    @Description("Retrieve all persons from the database")
    public Collection<Person> all_methodName() {
        return database.allPersons();
    }

    // To try out, see queries/mutation-create-person* files
    @Mutation(value = "create")
    @Description("Create a person")
    public Person create_methodName(Person person) {
        database.addPerson(person);
        return person;
    }

    // This effectively adds a "secretToken" field to the Person type. It is random and different each time it is requested.
    // To try out, see queries/generate-secret-tokens* files
    @Name("secretToken")
    public String generateSecretToken(@Source Person person,
                                      @DefaultValue("true")
                                      @Name("maskFirstPart") boolean maskFirstPart) {
        String uuid = UUID.randomUUID().toString();
        if (maskFirstPart) {
            return uuid.substring(0, uuid.length() - 4).replaceAll("[A-Za-z0-9]", "*")
                    + uuid.substring(uuid.length() - 4);
        } else {
            return uuid;
        }
    }

}

File: graphql/graphql-quarkus-kotlin/src/main/kotlin/org/example/graphql/api/PeopleApi.kt
package org.example.graphql.api

import org.eclipse.microprofile.graphql.*
import org.example.graphql.model.Person
import java.util.*
import javax.enterprise.context.ApplicationScoped

@GraphQLApi
@ApplicationScoped
class PeopleApi {

    // To try out, see queries/query-all-persons* files
    @Query(value = "all")
    @Description("Retrieve all persons from the database")
    fun all_methodName(): Collection<Person> {
        return PeopleDatabase.getAll()
    }

    // To try out, see queries/mutation-create-person* files
    @Mutation(value = "create")
    @Description("Create a person")
    fun create_methodName(person: Person): Person {
        PeopleDatabase.add(person)
        return person
    }

    // This effectively adds a "secretToken" field to the Person type. It is random and different each time it is requested.
    // To try out, see queries/generate-secret-tokens* files
    @Name("secretToken")
    fun generateSecretToken(@Source person: Person?,
                            @DefaultValue("true") @Name("maskFirstPart") maskFirstPart: Boolean): String {
        val uuid = UUID.randomUUID().toString()
        return if (maskFirstPart) {
            (uuid.substring(0, uuid.length - 4).replace("[A-Za-z0-9]".toRegex(), "*")
                    + uuid.substring(uuid.length - 4))
        } else {
            uuid
        }
    }

}